File: hyeyoung/week5/개인정보수집유효기간.java
package hyeyoung.week5;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class 개인정보수집유효기간 {
}

class Solution1 {
    public int[] solution(String today, String[] terms, String[] privacies) {
        List<Integer> answer = new ArrayList<>();
        Map<String, Integer> period = new HashMap<>();

        int date = getTotalDate(today);

        for(String t: terms){
            String[] term = t.split(" ");
            period.put(term[0], Integer.valueOf(term[1]));
        }

        for(int i = 0; i < privacies.length; i++){
            String[] privacy = privacies[i].split(" ");
            if(getTotalDate(privacy[0]) + (period.get(privacy[1]) * 28) <= date) {
                answer.add(i + 1);
            }
        }

        int[] intArray = new int[answer.size()];
        for (int i = 0; i < answer.size(); i++) {
            intArray[i] = answer.get(i);
        }

        return intArray;
    }

    private int getTotalDate(String today){
        String[] date = today.split("\\.");

        int year = Integer.parseInt(date[0]);
        int month = Integer.parseInt(date[1]);
        int day = Integer.parseInt(date[2]);

        return (year * 12 * 28) + (month * 28) + day;
    }
}

File: seoyeon/week5/개인정보 수집 유효기간.py
def solution(today, terms, privacies):
    today = list(map(int, today.split('.'))) # [0]연, [1]월, [2]일
    terms = {i[0]: int(i[2:])for i in terms}
    
    # 일 단위로 통합하기
    today = today[0] * 28 * 12 + today[1] * 28 + today[2] 
    
    answer = []
    for i, pri in enumerate(privacies):
        day, kind = pri.split()
        month = terms[kind]
        day = list(map(int, day.split('.')))
        day = day[0] * 28 * 12 + day[1] * 28 + day[2]-1 + month * 28 # 일 단위로 통합
        
        if day < today:
            answer.append(i+1)
                    
    return answer